torch/distributed/_op_stubs.py

```diff
@@ -0,0 +1,172 @@
+import torch.library
+
+from ._distributed_c10d import HAS_DISTRIBUTED
+
+
+# NB: This is separate from _distributed_c10d because the other module is
+# imported too early before operator registrations are ready.  Import this
+# from all modules that interact with operators.
+
+if not HAS_DISTRIBUTED:
+    # Define missing c10d operators
+    with torch.library._scoped_library("c10d", "DEF") as lib_def:
```

Comment:
This shouldn't be a scoped lib but a regular library if we want to def to remain right?

Reply:
I find this very confusing, because I see uses of scoped library that seemingly seem to imply that it is not gc'ed at the end. And empirically these library defs were enough to stop errors

```
with torch.library._scoped_library("aten", "IMPL") as aten:
    aten.impl("_nested_get_jagged_dummy", _nested_get_jagged_dummy, "CPU")
    aten.impl("_nested_get_jagged_dummy", _nested_get_jagged_dummy, "CUDA")
    aten.impl("_nested_get_jagged_dummy", _nested_get_jagged_dummy, "Meta")
```